### PR TITLE
FixBoolControl and fixBookUi revisions

### PR DESCRIPTION
**Pull request by johngalvin09**
I apologise for the multi branches, but this should have everything.

### CODE REVIEW

File: src/FixBookControl.java [hunk 1]
```diff
@@ -1,72 +1,69 @@
...
-
-
-
-
-
-}
+public class FixBookControl {
+
+    private FixBookUi userInterface;
+    private enum fixBookControlState { INITIALISED, READY, FIXING }
```
Comment (MethodicalMadness) [inline src/FixBookControl.java]: enums names start with a capital

File: src/FixBookControl.java [hunk 2]
```diff
@@ -1,72 +1,69 @@
...
+
+
+    public FixBookControl() {
+        //this should probably be a getInstance() per Jims notes. LIBRARY is not under my control, so request update?
+        this.library = library.INSTANCE();
+        state = fixBookControlState.INITIALISED;
+    }
+
+
+    public void setUI(FixBookUi newUserInterface) {
```
Comment (MethodicalMadness) [inline src/FixBookControl.java]: setUI needs to be setUi

File: src/FixBookUi.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+    public void run() {
+        output("Fix Book Interface\n");
+        while (true) {
+            switch (state) {
+            case READY:
+                String bookIdentifier = input("Scan Book ( <enter> completes): ");
+                if (bookIdentifier.length() == 0) {
+                    fixBookControl.scanningComplete();
+                }
```
Comment (MethodicalMadness) [inline src/FixBookUi.java]: else needs to be on preceding close braces, } else {

File: src/FixBookUi.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                else {
+                    try {
+                        int bookId = Integer.valueOf(bookIdentifier);
+                        fixBookControl.bookScanned(bookId);
+                    }
+                    catch (NumberFormatException exception) {
```
Comment (MethodicalMadness) [inline src/FixBookUi.java]: catch needs to be on preceding close braces, } catch {

File: src/FixBookUi.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+                    output("This is not a valid bookId");
+                    }
+                }
+                break;
+
```
Comment (MethodicalMadness) [inline src/FixBookUi.java]: remove empty line

File: src/FixBookUi.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+            case FIXING:
+                String fixBookAnswer = input("Would you like to fix Book? (Y/N) : ");
+                boolean fix = false;
+                if (fixBookAnswer.toUpperCase().equals("Y")) {
+                    fix = true;
+                }
+                fixBookControl.fixBook(fix);
+                break;
+
```
Comment (MethodicalMadness) [inline src/FixBookUi.java]: remove empty line

File: src/FixBookUi.java [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+            case COMPLETED:
+                output("The book/s have been fixed.");
+                return;
+
```
Comment (MethodicalMadness) [inline src/FixBookUi.java]: remove empty line

File: src/FixBookUi.java [hunk 8]
```diff
@@ -0,0 +1,83 @@
+import java.util.Scanner;
+
+
+public class FixBookUi {
+
+    public enum fixBookUserInterfaceState { INITIALISED, READY, FIXING, COMPLETED }
+
```
Comment (MethodicalMadness) [inline src/FixBookUi.java]: remove empty line between variables

File: src/FixBookControl.java (hunk 1 above)
Comment (johngalvin09) [reply src/FixBookControl.java]: Duly noted. I will rectify this

**Pull request merged**